pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -313,16 +313,16 @@ func (h *AutoscalersController) deleteAutoscaler(obj interface{}) {
 		return
 	}
 
-	autoscaler, ok := tombstone.Obj.(*autoscalingv2.HorizontalPodAutoscaler)
+	deletedHPA, ok = tombstone.Obj.(*autoscalingv2.HorizontalPodAutoscaler)
 	if !ok {
 		log.Errorf("Tombstone contained object that is not an Autoscaler: %#v", obj)
 		return
 	}
 
-	log.Debugf("Deleting Metrics from HPA %s/%s", hpa.Namespace, hpa.Name)
-	toDelete := h.hpaProc.ProcessHPAs(autoscaler)
+	log.Debugf("Deleting Metrics from HPA %s/%s", deletedHPA.Namespace, deletedHPA.Name)
+	toDelete := hpa.Inspect(deletedHPA)
```

Comment:
This is the purpose for the refactor. Since the hpa is deleted we don't want to make calls to Datadog.

`h.hpaProc.ProcessHPAs(autoscaler)`will make calls to Datadog but `hpa.Inspect(deletedHPA)` does not.